contrib/dockerd-rootless-setuptool.sh

```diff
@@ -313,7 +313,14 @@ install_systemd() {
 	fi
 	(
 		set -x
-		systemctl --user --no-pager --full status "${SYSTEMD_UNIT}"
+		if ! systemctl --user --no-pager --full status "${SYSTEMD_UNIT}"; then
+			set +x
```

Comment:
I think this `if` and `set +x` are going to conflate the output a bit, so I wonder if it would make more sense to move these outside the subshell, ala:

```bash
	if ! (
		set -x
		systemctl --user --no-pager --full status "${SYSTEMD_UNIT}"
		DOCKER_HOST="unix://$XDG_RUNTIME_DIR/docker.sock" $BIN/docker version
		systemctl --user enable "${SYSTEMD_UNIT}"
	); then
		if journalctl -q --user --unit "${SYSTEMD_UNIT}" -n 20 | grep -qF '/run/xtables.lock: Permission denied'; then
			INFO 'Failure likely related to https://github.com/moby/moby/issues/41230'
			INFO '`sudo dnf install -y policycoreutils-python-utils && sudo semanage permissive -a iptables_t` may work as a workaround.'
		fi
		exit 1
	fi
```

Reply:
This doesn't seem to catch failure in the subshell


```console
$ if !(set -eux; echo "should be printed 1"; false; echo "should not be printed"); then echo "should be printed 2"; fi
+ echo 'should be printed 1'
should be printed 1
+ false
+ echo 'should not be printed'
should not be printed
```

Reply:
`if !(set -x; echo "should be printed 1" && false && echo "should not be printed"); then echo "should be printed 2"; fi` works as expected, but looks more complicated